modmail/bot.py

```diff
@@ -19,27 +23,115 @@ class ModmailBot(commands.Bot):
+            allowed_mentions=allowed_mentions,
+            command_prefix=prefix,
+            intents=intents,
+            **kwargs,
+        )
+
+    def run(self, *args, **kwargs) -> None:
         """
-        Start the bot.
 
-        This just serves to create the http session.
+        Start up our instance of the bot. Since this method is blocking, it must be called last.
+
+        This method does several things, it loads extensions and plugins,
+        and then executes the main task.
+
+        This method was copied from discord.py and modified to suit our needs.
         """
-        await self.create_session()
-        await super().start(*args, **kwargs)
+        loop = self.loop
+
+        try:
+            loop.add_signal_handler(signal.SIGINT, lambda: loop.stop())
+            # this one we may want to get rid of, depending on certain things, and just hard stop instead.
+            loop.add_signal_handler(signal.SIGTERM, lambda: loop.stop())
+        except NotImplementedError:
+            pass
+
+        async def runner() -> None:
+            try:
+                # create the aiohttp session
+                self.http_session = ClientSession(loop=self.loop)
+                self.logger.trace("Created ClientSession.")
+                # set start time to when we started the bot
+                self.start_time = arrow.utcnow()
+                # we want to load extensions before we log in, so that any issues in them are discovered
+                # before we connect to discord
+                self.load_extensions()
+                # next, we log in to discord, to ensure that we are able to connect to discord
+                await self.login(*args)
+                # now that we're logged in and gotten a connection, we load all of the plugins
+                self.load_plugins()
+                # alert the user that we're done loading everything
+                self.logger.notice("Loaded all extensions, and plugins. Starting bot.")
+                # finally, we enter the main loop
+                await self.connect(**kwargs)
+            finally:
+                if not self.is_closed():
+                    await self.close()
+
+        def stop_loop_on_completion(f) -> None:  # noqa: ANN001
+            loop.stop()
+
+        future = asyncio.ensure_future(runner(), loop=loop)
+        future.add_done_callback(stop_loop_on_completion)
+        try:
+            loop.run_forever()
+        except KeyboardInterrupt:
+            self.logger.info("Received signal to terminate bot and event loop.")
+        finally:
+            future.remove_done_callback(stop_loop_on_completion)
+            self.logger.info("Cleaning up tasks.")
+            _cleanup_loop(loop)
+
+        if not future.cancelled():
+            try:
+                return future.result()
+            except KeyboardInterrupt:
+                # I am unsure why this gets raised here but suppress it anyway
+                return None
```

Comment:
Per discussion, this needs to be moved out of here so we don't end up God Objecting our way into misery.